### ISSUE TITLE
Relative paths not working in a playlist on linux

### ISSUE THREAD

**Issue by cyberconan** [libretro/RetroArch#17024]
## Description
Relative paths not working in a playlist on linux, only absolute paths.

### Expected behavior
Launch a game from a playlist with relative path.

### Actual behavior
The game is not launched.

### Steps to reproduce the bug
1. Edit a playlist created previously with RetroArch interface than works well.
2. Replace absolute path like "/home/karl/.config/retroarch/roms/fbneo/1941.zip" with relative path like ":/roms/fbneo/1941.zip".
3. When try to launch the game from playlist, it not load

### Bisect Results
I detected this when I was trying to prepare some playlist for my SteamDeck from my desktop installation.
I tried other relative path options like  ":/cores/../roms/fbneo/1941.zip", "roms/fbneo/1941.zip" or "~/.config/retroarch/roms/fbneo/1941.zip", but no one work.

Relative paths works perfect for core line: "core_path": ":/cores/fbneo_libretro.so".

### Version/Commit
1.19.1

### Environment information
- OS: ArchLinux
- Compiler:  Using official version in Arch Extra repository

**Comment by i30817** [comment]
Use the portable playlist feature to relocate playlists between systems or OSes boots, playlist only work with absolute paths on purpose because the current directory is a very relative thing for RetroArch, on OSes boots or even user changes.

**Comment by i30817** [comment]
Don't reuse retroarch.cfg across systems\oses\users either. That's a fine way to end with a lot of funky bugs because that file has many os specific settings or even user specific paths.

**Comment by cyberconan** [comment]
Hi! Relative paths like ":/roms" works perfect in playlists with Windows file system. The problem with linux is than "/" is concatenated always before the path. Regards!
![relative-path](https://github.com/user-attachments/assets/743ab229-1c05-452b-86a4-b36db1b74ed4)

**Comment by i30817** [comment]
Like I wrote before, relative paths aren't part of the paths of roms on purpose. The reason is that it would sabotage portable playlists because a relative path is relative to the current system, and portable playlists work by memorizing a ancestor path of the game directories in the playlists to compare to another system browser path and when they different, change them. A relative path would never match, and so a part with them would never be transformed, so the portable playlists setting would do nothing.

**Comment by i30817** [comment]
I'm almost certain that what you're trying to do with relative paths can be done with portable playlists instead.

**Comment by cyberconan** [comment]
@i30817, what you propose is to avoid the problem by using another great functionality of RetroArch, but not solve the issue I reported.

For those of us who move games on an SD from a laptop with Linux to Steam Deck, I think it is infinitely easier to copy the roms to a folder in the RetroArch root directory because the playlist would always work and without doing anything else.

**Comment by i30817** [comment]
That's a great way to get a corrupted retroarch.cfg. As I mentioned, do not, ever share them across OSes. That feature is what is intended to share playlists and games, so copy or use a external drive, it's there to do what you want, but don't say you weren't warned when you get low res icons or failing GPU driver settings when you try to copy the whole RetroArch folder across systems.

**Comment by cyberconan** [comment]
You are wrong in your approach to what I do. I don't copy the entire retroarch folder from my laptop to the SD card, I only copy some folders such as playlists and roms, the rest is already provided by Steam with the official RetroArch installation.

**Comment by i30817** [comment]
Then just use portable playlists feature. Instead of trying to ask devs into allowing relative paths - a alternative to that - that was rejected _because_ it breaks that feature and because it requires things that the RA interface does not accept. Such as adding yet another option to both scanners, on different places to collect relative to the Ra folder paths...

**Comment by cyberconan** [comment]
As I said before, I have only reported an issue that is only present on Linux because on Windows relative paths work fine. If it is never solved, nothing happens, Linux users will survive by looking for, collaborating and using alternatives as always. ;)

Thank you @i30817 for your comments.

**Comment by i30817** [comment]
It's your time man, I don't think it will work out like you think though even if you get the 'limitation' removed. In fact, that the Linux code intentionally adds / always indicates to me a bug fix happened because something was going wrong with the very clumsy way Ra uses its own hadhoc relative path protocol. (:\)

**Comment by yhojann-cl** [comment]
I believe the restriction against using relative paths and the home character "\~" is completely unnecessary and ruins the user experience for something "believed to be unnecessary," when this is not the case. For example, I have the official download of the AppImage version for Linux, which comes compressed in a 7z file that includes a virtual home directory so that all configurations and ROMs are stored only in that directory and nowhere else. The configuration file correctly uses the path "~/.config/..." and works perfectly because it considers the virtual directory and not the home directory of the main host user. The problem is that playlists force the use of absolute paths, breaking the idea of ​​making RetroArch portable through AppImage. The playlists depend directly on the absolute path where I save the AppImage file and don't allow me to move it to another location, such as a USB stick or external hard drive, because the path would always change. This is more of a hindrance than a help. What's stopping them from allowing relative paths? Let RetroArch internally generate absolute paths. But if not, then what's the problem with people using short paths? I don't see any need to restrict or censor its use. It's like telling me what kind of controllers I should use for my game because someone thinks it will be better for me.
